src/models/LinkModel.ts

```diff
@@ -51,6 +51,19 @@ export class LinkModel extends BaseModel<LinkModelListener> {
 		});
 	}
 
+	clone(lookupTable) {
+		if(((lookupTable||{})[this.class]||{}).hasOwnProperty(this.id)) return lookupTable[this.class][this.id];
```

Comment:
would prefer this to be on separate lines with context braces to improve readability, I also don't think we need the 'hasOwnProperty', just a simple `lookupTable[this.class][this.id]` should suffice, especially if we give the lookupTable parameter a type.